docs/DeveloperGuide.md

```diff
@@ -18,12 +44,162 @@
 
 {Describe the value proposition: what problem does it solve?}
 
+
+## Implementations
+### Exit Command
+
+The `ExitCommand` inherits its properties from the abstract `Command` class.
+
+The command to exit the program is `exit`.
+If 'exit' is called, the program prints the exit message and terminates the program.
+
+This is how the ExitCommand works:
+
+1. The `main()` method in Main calls `run()` in Main. 
+
+2. `run()` will call `runCommandTillExit()`.
+
+3. The `ui` reads the user's input to retrieve
+the command via `ui.getUserInput()` and parses it through `commandParser.parseCommand()`.
+
+4. Within the `commandParser.parseCommand()`, some functions are also internally called.
+    1. `newCommand()` is called to identify the String received
+    2. `parseKeyword()` is called to split the command from the parameters.
+ 
+5. A new `ExitCommand()` is called.
+
+6. `ExitCommand` is returned to parseCommand().
+
+7. `ExitCommand` is returned to main().
+
+8. The Exit Command is now executed via `command.execute()`.
+
+9. `runCommandTillExit()` will now call `command.isExit()` which returns `true`.
+
+10. The loop is now broken and ui will call `ui.showEndingMessage()`.
+
+11. The program will now exit.
+
+<p align="center">
+    <img src="images/ExitCommandSequenceDiagram.png">
+    <br />
+    <i>Figure 2: Sequence Diagram for Bye Command</i>
+</p>
```

Comment:
Regarding this diagram, there is missing text at the arrow from :CommandParser to :ExitCommand.